### PR TITLE
Fix publish issues

### PR DESCRIPTION
**Pull request by ellismg**
Fix some issues that were not discovered until publishing ran for the new distros.

### CODE REVIEW

**Comment by ellismg** [comment]
I manually published fixed builds to test this.

File: build_projects/dotnet-host-build/PublishTargets.cs [hunk 1]
```diff
@@ -266,6 +266,11 @@ public static BuildTargetResult PublishCoreHostPackages(BuildTargetContext c)
         [Target]
         public static BuildTargetResult PublishSharedHostInstallerFileToAzure(BuildTargetContext c)
         {
+            if (CurrentPlatform.IsUbuntu && !CurrentPlatform.IsVersion("14.04"))
```
Comment (joshfree) [inline build_projects/dotnet-host-build/PublishTargets.cs]: is this a temp workaround?

Comment (ellismg) [inline build_projects/dotnet-host-build/PublishTargets.cs]: Yes.  We don't generate debs here yet.

Comment (joshfree) [inline build_projects/dotnet-host-build/PublishTargets.cs]: OK.  LGTM (to unblock publishing)

**Comment by ellismg** [comment]
The Fedora 23 build failed because the CI machines do not have tar on them.  I will update the images and deploy new builds.

**Comment by ellismg** [comment]
@dotnet-bot test Fedora23 x64 Debug Build please

**Pull request merged**